kernelci.org/content/en/docs/contrib.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "Contributing Guidelines"
+date: 2022-12-08T09:21:56Z
+draft: false
+weight: 2
+---
+
+KernelCI core project is open for contributions. Contributions may consist of
+adding new build, tests and device types as well as features and bugfixes for
+KernelCI core tools.
+The best way to contribute is to send a PR to [kernelci-core](https://github.com/kernelci/kernelci-core).
+When the PR is created, [KernelCI staging](https://kernelci.org/docs/instances/staging)
+instance takes care of updating the [staging.kernelci.org branch](https://github.com/kernelci/kernelci-core/tree/staging.kernelci.org).
+In general the branch is updated every 8h and a limited set of builds and tests
+is run on it. More detailed information about the logic behind staging runs can
+be found [here](https://kernelci.org/docs/instances/staging).
+
+There are several guidelines which can facilitate the PR review process.
+
+1. Make sure the PR is well described
+   1. Describe the purpose of the changes
+   2. Example use cases are welcome
+2. Attach staging build/test results when possible
```

Comment:
Use a permanent link for staging build/test results when possible, as current links get removed after few weeks.
maybe suggest pastebin ( like github gist )

Reply:
IMHO it is a bit hard to pastebin results from dashboard, where we have heavy ajax. Probably it is acceptable to have expiring link for new tests PR and just as confirmation that all works as designed.
There is very small chance we will need to review this test results.
But i agree that we need permanent links in case of discovered real bugs or commits.

Reply:
> where we have heavy ajax.
I'm talking mostly about logs that are on https://storage.staging.kernelci.org/ and https://storage.kernelci.org/ that are temporary

Reply:
just use something like wgetpastebin

Reply:
Staging builds and tests aren't permanent.

Things we can treat as permanent are mailing list archives, anything on GitHub (issues, PRs etc.) and in theory results on the production dashboard although that's more a grey area as they might disappear after a while too.

I think it's best if we can just mention some log excerpts in the commit messages directly inline, and use `Link:` for email threads like in the kernel git history.